### ISSUE TITLE
Split gem: a generic "concept" gem and IEV specific parsing

### ISSUE THREAD

**Issue by ronaldtse** [glossarist/iev-data#185]
From metanorma/stepmod-utils#14:

We need a Ruby gem to deal with the https://github.com/glossarist/concept-model model.

This will allow us to use the same basis for ingesting concepts as well exporting them.

This gem will be used in metanorma/metanorma-standoc#255 by @opoudjis .

**Comment by skalee** [comment]
Yes, I've noticed this one. Though I am thinking if we can do it in some meta language to avoid separate Ruby and JS implementations… Would be great I suppose, but I'm not sure if possible.

**Comment by skalee** [comment]
And I guess that I should close few other ones first. Unless this is really urgent.

**Comment by ronaldtse** [comment]
@skalee yes this is relatively urgent work, definitely more urgent than completion of parsing of IEV because that comes later. Thanks.

**Comment by skalee** [comment]
@ronaldtse, @opoudjis 

- Which concept features are more needed? These used in IEV or these in TC211?
- Any requirements regarding supported Ruby versions? Should I care about Ruby 2.4? Should I care about anything below 2.7?

**Comment by ronaldtse** [comment]
> Which concept features are more needed? These used in IEV or these in TC211?

We need all from both, because they are all real requirements. The authoritative model is https://github.com/glossarist/concept-model .

> Any requirements regarding supported Ruby versions? Should I care about Ruby 2.4? Should I care about anything below 2.7?

We have to support 2.4+ for now.

**Comment by skalee** [comment]
I wonder how to name this gem. `concepts`, `glossary_concepts, `glossarist_concepts`, `iso_concepts`, `glossarist` are all free, `concept` is taken.

**Comment by ronaldtse** [comment]
Maybe just `glossarist` for now?

**Comment by skalee** [comment]
Agreed. Two more questions:

- Any requirements regarding license? I am going to use BSD 2-clause, as Metanorma uses specifically this one.
- Should I create a new repo for this project or keep it in glossarist/concept-model in a subdirectory? I am going to do the former which is cleaner IMO, but maybe there are reasons to do otherwise?

**Comment by ronaldtse** [comment]
> * Any requirements regarding license? I am going to use BSD 2-clause, as Metanorma uses specifically this one.

Agree.

> * Should I create a new repo for this project or keep it in glossarist/concept-model in a subdirectory? I am going to do the former which is cleaner IMO, but maybe there are reasons to do otherwise?
Here's the new repo: https://github.com/glossarist/glossarist-ruby

> * Is it okay to support Ruby 2.4.4+, i.e. without Ruby 2.4.0-2.4.3? That would allow us to use [Zeitwerk](https://github.com/fxn/zeitwerk#supported-ruby-versions). Though I can live without it.

I think so.
